test/sqllogictest/transform/predicate_pushdown.slt

```diff
@@ -0,0 +1,84 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This file contains tests for the PredicatePushdown transform.
+
+statement ok
+CREATE TABLE x (a int not null, u int, b bool)
+
+statement ok
+CREATE TABLE y (a int not null)
+
+query T multiline
+EXPLAIN PLAN FOR SELECT b FROM (SELECT b, not(b) as neg FROM x) WHERE NOT(neg)
+----
+%0 =
+| Get materialize.public.x (u1)
+| Filter #2
+| Map null
+| Project (#2)
+
+EOF
+
+query T multiline
+EXPLAIN PLAN FOR SELECT b FROM (SELECT b, b = false as neg FROM x) WHERE NOT(neg)
+----
+%0 =
+| Get materialize.public.x (u1)
+| Filter (#2 != false)
+| Map null
+| Project (#2)
+
+EOF
+
+# Inline binary expressions.
+
+query T multiline
+EXPLAIN PLAN FOR
+  SELECT a FROM (SELECT a, a = 3 AS cond, u != 2 as cond2 FROM x) WHERE NOT(cond) AND NOT(cond2)
+----
+%0 =
+| Get materialize.public.x (u1)
+| Filter (#1 = 2), (#0 != 3)
+| Map false, null
+| Project (#0)
```

Comment:
This is surprising to see. The `LiteralLifting` transform is meant to get rid of these. I can imagine it is broken somehow, but we should chase that down.